loony/spec.nim

```diff
@@ -1,17 +1,32 @@
-import std/atomics
+import std/[atomics, math]
 
 const
   loonyNodeAlignment {.intdefine.} = 11
-  loonySlotCount {.intdefine.} = 1024
+  loonySlotCount* {.intdefine.} = 1024
 
   loonyIsolated* {.booldefine.} = false  ## Indicate that loony should
   ## assert that all references passing through the queue have a single
   ## owner.  Note that in particular, child Continuations have cycles,
   ## which will trigger a failure of this assertion.
 
+  loonyRotate* {.booldefine.} = true ## Indicate that loony should rotate
+  ## the slots in the queue to avoid contention on the same cache line.
+  ## This is useful when the queue is shared between multiple threads.
+  ## Note that this will only work if the number of slots is a power of 2.
+
+when loonyRotate:
+  # Impl dynamic cache line size detection
+  const
+    cacheLineSize = 64
+    lShiftBits* = int log2(float cacheLineSize)
+    rShiftBits* = int(log2(float loonySlotCount)) - lShiftBits
+
 static:
   doAssert (1 shl loonyNodeAlignment) > loonySlotCount,
     "Your LoonySlot count exceeds your alignment!"
+  when loonyRotate:
+    doAssert (loonySlotCount and (loonySlotCount - 1)) == 0,
+      "LoonySlot count must be a power of 2!"
```

Comment:
This won't catch nimpleton values of 0, 1.
The error message should include the symbol `loonySlotCount`, right?